src/frontend/src/hooks/UsePluginPanels.tsx

```diff
@@ -0,0 +1,129 @@
+
+export function usePluginPanels({
+  instance,
+  model,
+  id
+}: {
+  instance?: any;
+  model?: ModelType | string;
+  id?: string | number | null;
+}): PanelType[] {
+  const host = useLocalState.getState().host;
+  const navigate = useNavigate();
+  const user = useUserState();
+  const theme = useMantineTheme();
+  const globalSettings = useGlobalSettingsState();
+
+  const pluginPanelsEnabled: boolean = useMemo(
+    () => globalSettings.isSet('ENABLE_PLUGINS_INTERFACE'),
+    [globalSettings]
+  );
+
+  // API query to fetch initial information on available plugin panels
+  const { data: pluginData } = useQuery({
+    enabled: pluginPanelsEnabled && !!model && id !== undefined,
+    queryKey: [model, id],
+    queryFn: async () => {
+      if (!pluginPanelsEnabled || !model) {
+        return Promise.resolve([]);
+      }
+
+      return api
+        .get(apiUrl(ApiEndpoints.plugin_panel_list), {
+          params: {
+            target_model: model,
+            target_id: id
+          }
+        })
+        .then((response: any) => response.data)
+        .catch((error: any) => {
+          console.error('Failed to fetch plugin panels:', error);
+          return [];
+        });
+    }
+  });
+
+  // Cache the context data which is delivered to the plugins
+  const contextData: PluginContext = useMemo(() => {
+    return {
+      model: model,
+      id: id,
+      instance: instance,
+      user: user,
+      host: host,
+      api: api,
+      navigate: navigate,
+      theme: theme
+    };
+  }, [instance, model, theme, id, user, host, api, navigate]);
+
+  // Track which panels are hidden: { panelName: true/false }
+  // We need to memoize this as the plugins can determine this dynamically
+  const [panelState, setPanelState] = useState<Record<string, boolean>>({});
+
+  // Clear the visibility cache when the plugin data changes
+  // This will force the plugin panels to re-calculate their visibility
+  useEffect(() => {
+    pluginData?.forEach((props: PluginPanelProps) => {
+      const identifier = identifierString(
+        `plugin-panel-${props.plugin}-${props.name}`
+      );
+
+      // Check if the panel is hidden (defaults to true until we know otherwise)
+      isPluginPanelHidden({
+        pluginProps: props,
+        pluginContext: contextData
+      }).then((result) => {
+        setPanelState((prev) => ({ ...prev, [identifier]: result }));
+      });
+    });
+  }, [pluginData, contextData]);
+
+  const pluginPanels: PanelType[] = useMemo(() => {
+    return (
+      pluginData?.map((props: PluginPanelProps) => {
+        const iconName: string = props.icon || 'plugin';
+        const identifier = identifierString(
+          `plugin-panel-${props.plugin}-${props.name}`
+        );
+        const isHidden: boolean = panelState[identifier] ?? true;
+
+        return {
+          name: identifier,
+          label: props.label,
+          icon: <InvenTreeIcon icon={iconName as InvenTreeIconType} />,
```

Comment:
Q: Do you see a way here so we can use any tabler icon here? Not just the picked ones, available through the `InvenTreeIcon` component without adding much to the current bundle build size. Maybe we can use the APIIcon component here as a fallback?

Reply:
Yes, there is probably a way to re-use the existing icon framework! If you are ok with it, I'll leave that for a future work